lib/pkcs11h-certificate.c

```diff
@@ -1316,9 +1333,9 @@ pkcs11h_certificate_sign (
 
 	_PKCS11H_DEBUG (
 		PKCS11H_LOG_DEBUG2,
-		"PKCS#11: pkcs11h_certificate_sign entry certificate=%p, mech_type=%ld, source=%p, source_size="P_Z", target=%p, *p_target_size="P_Z"",
+		"PKCS#11: pkcs11h_certificate_sign_ex entry certificate=%p, mech_type=%ld, source=%p, source_size="P_Z", target=%p, *p_target_size="P_Z"",
 		(void *)certificate,
-		mech_type,
+		mech->mechanism,
```

Comment:
once you add the assert above, you can do this :)

Reply:
Good catch!